### ISSUE TITLE
Changing the alternative of a parameter value associated with a parameter list with non-numeric values causes an error

### ISSUE THREAD

**Issue by DillonJ** [spine-tools/Spine-Toolbox#2330]
See attached minimal example. To reproduce in the attached minimal example, change the alterative of the testp2 parameter value.

From any other DB:

- Create a new parameter list with non-numeric values (e.g. "abc" and "def").
- Now associate that list with an object parameter definition
- Create a parameter value for an object selecting one of the items from the newly created list (e.g. "abc")
- Create a new alternative called "dummy"
- Change the alternative of the above non-numerical parameter value to "dummy".

You now get an error along the lines of 

```
From test.sqlite:
Invalid value '3.0' - it should be one from the parameter value list: '"abc"', '"def"'.
```

[test.zip](https://github.com/spine-tools/Spine-Toolbox/files/12716507/test.zip)

Tested in 0.7, haven't tried in 0.8

**Comment by jkiviluo** [comment]
Great to have this reproduced. I've seen the error message many times, but haven't figured out the pattern. I don't think anything goes wrong other than the message, right? It's confusing though.

Let's see if it's in 0.8, fix there first and roll it to 0.7 too.

**Comment by PiispaH** [comment]
This doesn't seem to happen in 0.8 anymore. Probably not worth it to fix in 0.7 @soininen @jkiviluo ?

**Comment by jkiviluo** [comment]
I'm ok if it's not fixed in 0.7, since it doesn't corrupt data (although users do see it from time to time). How about @DillonJ?

**Comment by PiispaH** [comment]
Doesn't replicate in 0.8.
